evaluator/builtin_string.go

```diff
@@ -180,103 +180,114 @@ func builtinUpper(args []types.Datum, _ context.Context) (d types.Datum, err err
 }
 
 // See: https://dev.mysql.com/doc/refman/5.7/en/string-comparison-functions.html
-func builtinStrcmp(args []interface{}, _ context.Context) (interface{}, error) {
-	if args[0] == nil || args[1] == nil {
-		return nil, nil
+func builtinStrcmp(args []types.Datum, _ context.Context) (d types.Datum, err error) {
```

Comment:
Put `d.SetNull()` here, then if `err != nil` is true , we don't need to set it again?